### PR TITLE
[MB-5153 // MB-5350 // MB-5352 // MB-5353] Fix final EDI generation issues part 2 // Fix for FA1 and FA2 // Add Buyer and Seller // Update unit tests

### PR DESCRIPTION
**Pull request by harshamurthy**
## Description

There were errors reported by Syncada in our EDI file. This PR
- moves FA1 and FA2 from the header to each segment
- adds N1 segments for buyer and seller names
- updates unit tests in accordance with the changes above

## Setup

Start the server with e2e basic data

```sh
make db_dev_e2e_populate
make server_run
```

Save following payload to a new file `new_payment_request.json`

```json
{
  "body": {
    "isFinal": false,
    "moveTaskOrderID": "9c7b255c-2981-4bf8-839f-61c7458e2b4d",
    "serviceItems": [
      {
        "id": "94bc8b44-fefe-469f-83a0-39b1e31116fb"
      },
      {
        "id": "eee4b555-2475-4e67-a5b8-102f28d950f8"
      },
      {
        "id": "6431e3e2-4ee4-41b5-b226-393f9133eb6c"
      },
      {
        "id": "fd6741a5-a92c-44d5-8303-1d7f5e60afbf"
      }
    ]
  }
}
```

Create a payment request:

```sh
❯ go run ./cmd/prime-api-client --insecure create-payment-request --filename new_payment_request.json | jq '.paymentRequestNumber'
```

Then generate the EDI file

```sh
❯ go run ./cmd/generate-payment-request-edi/ --payment-request-number [Enter Payment Request Number]
```

Compare your result to the expected EDI below.
There should now be two additional segments in the header:
```
N1*BY*BuyerOrganizationName*92*LKNQ
N1*SE*SellerOrganizationName*2*BLKW
```
 there should be and `FA1` and `FA2` in each service item segment:
```
HL*1**9
N9*PO*3351-1123-be758fac
L5*1*DDP*TBD*D
L0*1***1349.000*B******L
L1*1*1349*LB*10835
FA1*ZZ
FA2*TA*F8E1
HL*2**9
N9*PO*3351-1123-ba793d9c
L5*2*DSH*TBD*D
L0*2*413.000*DM*1349.000*B******L
L1*2*1349*LB*506066
FA1*ZZ
FA2*TA*F8E1
HL*3**9
N9*PO*3351-1123-3361fb47
L5*3*DLH*TBD*D
L0*3*373.000*DM*1349.000*B******L
L1*3*1349*LB*1221717
FA1*ZZ
FA2*TA*F8E1
```

```
ISA*00*0084182369*00*0000000000*ZZ*MILMOVE        *12*8004171844     
*201023*2114*U*00401*100001280*0*T*|
GS*SI*MILMOVE*8004171844*20201023*2114*100001280*X*004010
ST*858*0001
BX*00*J*PP*3351-1123b*BLKW**4
N9*CN*3351-1123-1b
N9*CT*TRUSS_TEST
N9*1W*Leo, Spacemen
N9*ML*E_1
N9*3L*ARMY
G62*10*20201023
G62*76*20201022
G62*86*20200316
N1*BY*BuyerOrganizationName*92*LKNQ
N1*SE*SellerOrganizationName*2*BLKW
N1*ST**10*CNNQ
N3*Fort Gordon
N4*Augusta*GA*30813*US
PER*CN**TE*706-791-4184
N1*SF*Uoe1WjuUjU*10*LKNQ
N3*987 Other Avenue*P.O. Box 1234
N4*Des Moines*IA*50309*US
PER*CN**TE*(510) 555-5555
HL*1**9
N9*PO*3351-1123-be758fac
L5*1*DDP*TBD*D
L0*1***1349.000*B******L
L1*1*1349*LB*10835
FA1*ZZ
FA2*TA*F8E1
HL*2**9
N9*PO*3351-1123-ba793d9c
L5*2*DSH*TBD*D
L0*2*413.000*DM*1349.000*B******L
L1*2*1349*LB*506066
FA1*ZZ
FA2*TA*F8E1
HL*3**9
N9*PO*3351-1123-3361fb47
L5*3*DLH*TBD*D
L0*3*373.000*DM*1349.000*B******L
L1*3*1349*LB*1221717
FA1*ZZ
FA2*TA*F8E1
HL*4**9
N9*PO*3351-1123-589423ed
L5*4*FSC*TBD*D
L0*4*373.000*DM*1349.000*B******L
L1*4*1349*LB*362
FA1*ZZ
FA2*TA*F8E1
L3*1349.000*B***1738980
SE*50*0001
GE*1*100001280
IEA*1*100001280
```

## Code Review Verification Steps

* [X] Request review from a member of a different team.
* [X] Have the Jira acceptance criteria been met for this change?

## References

* [MB-5153](https://dp3.atlassian.net/browse/MB-5153) for this change
* [MB-5350](https://dp3.atlassian.net/browse/MB-5350) for this change
* [MB-5352](https://dp3.atlassian.net/browse/MB-5352) for this change
* [MB-5353](https://dp3.atlassian.net/browse/MB-5353) for this change

### CODE REVIEW

**Review by pearl-truss — APPROVED** [review]
Looks good! Is the plan to keep this branch and PR open until you add the code from other subtasks to here?

**Review by gidjin — COMMENTED** [review]
Changes look good, code works. As expected tests are broken, so those will need to be fixed before the PR can be merged.

File: pkg/services/invoice/ghc_payment_request_invoice_generator.go [hunk 1]
```diff
@@ -324,6 +325,46 @@ func (g ghcPaymentRequestInvoiceGenerator) createG62Segments(paymentRequestID uu
...
+
+	var err error
+	var originDutyStation models.DutyStation
+	if orders.OriginDutyStation.ID != uuid.Nil {
+		originDutyStation, err = models.FetchDutyStation(g.db, *orders.OriginDutyStationID)
+		if err != nil {
+			return []edisegment.Segment{}, services.NewInvalidInputError(orders.OriginDutyStation.ID, err, nil, "unable to find new duty station")
+		}
+	} else {
+		return []edisegment.Segment{}, services.NewBadDataError("Invalid Order, must have OriginDutyStation")
```
Comment (gidjin) [inline pkg/services/invoice/ghc_payment_request_invoice_generator.go:339]: As I understand from https://github.com/transcom/mymove/wiki/API-Errors#http-status-codes we should not use `BadDataError` for this type of issue. It should be `ConflictError`. Though I realize we are using BadDataError else where in this file when we shouldn't be. So maybe create a chore to clean those up.

File: pkg/services/invoice/ghc_payment_request_invoice_generator.go [hunk 2]
```diff
@@ -324,6 +325,46 @@ func (g ghcPaymentRequestInvoiceGenerator) createG62Segments(paymentRequestID uu
 	return g62Segments, nil
 }
 
+func (g ghcPaymentRequestInvoiceGenerator) createBuyerAndSellerOrganizationNamesSegments(paymentRequestID uuid.UUID, orders models.Order) ([]edisegment.Segment, error) {
+	buyerAndSellerOrganizationNames := []edisegment.Segment{}
+
+	var err error
+	var originDutyStation models.DutyStation
+	if orders.OriginDutyStation.ID != uuid.Nil {
```
Comment (gidjin) [inline pkg/services/invoice/ghc_payment_request_invoice_generator.go:333]: Small nit we're doing this basically twice in this file now in different ways. It's worth making the checks the same, possibly even worth pulling them both up to earlier in the file and only checking and loading it once.

File: pkg/services/invoice/ghc_payment_request_invoice_generator.go [hunk 3]
```diff
@@ -324,6 +325,46 @@ func (g ghcPaymentRequestInvoiceGenerator) createG62Segments(paymentRequestID uu
...
+		originDutyStation, err = models.FetchDutyStation(g.db, *orders.OriginDutyStationID)
```
Comment (pearl-truss) [inline pkg/services/invoice/ghc_payment_request_invoice_generator.go:334]: I think this is causing the 'nil pointer derefence' errors in the server test. The nil check on line 333 is for orders.OriginDutyStation.ID but this line is using orders.OriginDutyStationID which doesn't have a nil check.

File: pkg/services/invoice/ghc_payment_request_invoice_generator.go (hunk 1 above)
Comment (harshamurthy) [reply pkg/services/invoice/ghc_payment_request_invoice_generator.go:340]: Created a chore!

**Review by pearl-truss — APPROVED** [review]
Looks good, thanks for making the changes!

**Pull request merged**